### PR TITLE
expression: remove type assertion on `types.DatumRow`.

### PR DESCRIPTION
**Pull request by coocood**
So we can change the implementation of `types.Row` from `types.DatumRow` to `chunk.Row`.

### CODE REVIEW

File: expression/column.go [hunk 1]
```diff
@@ -194,48 +195,58 @@ func (col *Column) GetType() *types.FieldType {
...
+		val := row.GetDatum(col.Index, col.RetType)
+		if val.IsNull() {
+			return 0, true, nil
+		}
 		res, err := val.ToFloat64(sc)
 		return res, err != nil, errors.Trace(err)
 	}
-	return val.GetFloat64(), false, nil
+	if col.GetType().Tp == mysql.TypeFloat {
+		val, isNull := row.GetFloat32(col.Index)
```
Comment (winkyao) [inline expression/column.go]: need distinguish float32?

Comment (coocood) [reply expression/column.go]: float32 is store in 4 bytes in chunk

**Comment by winkyao** [comment]
Please fix conflict

File: util/types/row.go [hunk 2]
```diff
@@ -53,13 +55,21 @@ type Row interface {
 
 	// GetJSON returns the JSON value and isNull with the colIdx.
 	GetJSON(colIdx int) (json.JSON, bool)
+
+	// GetDatum returns a Datum with the colIdx and field type.
+	GetDatum(colIdx int, tp *FieldType) Datum
```
Comment (zz-jason) [inline util/types/row.go]: can we use `*Datum ` as return type ?

Comment (coocood) [reply util/types/row.go]: `*Datum` need to allocate memory.

**Comment by coocood** [comment]
@zz-jason 
PTAL

**Pull request merged**